torch/csrc/jit/codegen/cuda/scheduler/registry.cpp

```diff
@@ -1219,6 +1219,22 @@ class ReductionScheduler : public SchedulerEntry {
       return false;
     }
 
+    // Check that inputs of all select ops are fusion inputs
+    for (auto select : ir_utils::getSelectOps(fusion)) {
+      if (!select->input(0)->isFusionInput()) {
+        scheduler_debug_utils::canScheduleRejectReason(
+            ScheduleHeuristic::Reduction,
+            "Inputs of SelectOp must be fusion input.");
+        return false;
+      }
+      if (select->input(0)->uses().size() > 1) {
```

Comment:
Maybe this restriction can be lifted, at least for the case where you have multiple select ops at the same dim.